Sale by packaging consolidate changes

simahawk opened the pull request:
Integrate https://github.com/OCA/sale-workflow/pull/1231/

simahawk committed:
sale_by_packaging: no pkging override if already set

Before this change the packaging was always overridden
hence the user had not control on the packaging to be used on SO line.

simahawk committed:
sale_by_packaging: skip auto assign when forcing uom qty

When the product_uom_qty is computed from inverse method of packaging qty
the packaging is already there and there's no need to auto-assign.

This also prevents skipping direct writes on prod uom qty or other keys
other than product_packaging.

simahawk committed:
sale_by_packaging: drop useless check on pkging multiple qty

This check is already done by core sale module.
Tests where not checkingt for that specific exception.

TDu commented on sale_by_packaging/models/sale_order_line.py:
```diff
@@ -76,44 +76,29 @@ def _force_qty_with_package(self):
-        if vals.get("product_packaging") or not any(
-            fname in vals for fname in fields_to_check
-        ):
+        if vals.get("product_packaging") or self.env.context.get("_skip_auto_assign"):
```
```suggestion
        if "product_packaging" in vals.keys() or self.env.context.get("_skip_auto_assign"):
```
The product_packaging value can be False.

simahawk replied:
the branch is on camptocamp: you can push to it. Or, merge it and change what you want in your branch :wink:

TDu replied:
But what is the reason behind those changes ?
Because they break 4 out of 7 tests
Where they included before ?

TDu reviewed (commented):
With this changes and looking at the broken unit tests. It seems I can not set the product_packaging_id to False after it has been set.

TDu committed:
sale_by_packaging: Fix tests and ask question

The product_product_packaging_qty on sale.order.line should always be an
integer, no ?

Removing the test on the onchange has it has been removed.
